tests/readrows_test.go

```diff
@@ -892,8 +890,13 @@ func TestReadRows_Retry_WithRoutingCookie_MultipleErrorResponses(t *testing.T) {
 	// 4a. Verify that the read succeeds
 	checkResultOkStatus(t, res)
 	assert.Equal(t, 2, len(res.GetRows()))
-	assert.Equal(t, "row-01", string(res.Rows[0].Key))
```

Comment:
This code was reading past the end of the array and segfaulting when the length was less than  2.

Reply:
wouldn't that fail on line 894?

Reply:
Do you mean old line 894, `assert.Equal(t, 2, len(res.GetRows()))`, or new line 894 `assert.Equal(t, "row-01", string(res.Rows[0].Key))`?